pdfly/sign.py

```diff
@@ -0,0 +1,42 @@
+"""Sign a PDF file with a PKCS#12 certificate, retaining original formatting and adding a signature page."""
+
+from pathlib import Path
+from fpdf import FPDF
+import fitz  # PyMuPDF for precise page content extraction
```

Comment:
This dependency is a new dependency, that `pdfly` currently does not have.

Why introducing it?

The feature-request issue suggested to use `fpdf2`, that is already a dependency of `pdfly`, and also provide signing ability: https://github.com/py-pdf/pdfly/issues/71

Reply:
Because I want to keep the original pdf content and add a signature, fpdf2 always needs to update an empty temporary pdf，i have no idea to slove it.   :(

Reply:
> fpdf2 always needs to update an empty temporary pdf，i have no idea to slove it. :(

You could share the code that you used, and the details of the error you face, and I'll to help you solve it 🙂